### ISSUE TITLE
Support caret block style syntax

### ISSUE THREAD

**Issue by clitetailor** [clitetailor/cursornext#1]
Implement the `diagram` method that convert a caret diagram into the equivalent caret syntax. For example the following code:

```ts
t.diagram`
    1 | Hello, World!
      | ^      ^
      | |      cursor
      | marker
    2 | This is an example of how caret diagram and caret syntax work.
`
```

would returns:

```
🌵(marker)Hello, 🌵(cursor)World!
This is an example of how caret diagram and caret syntax work.
```

For documentation, checkout [Writing Test](https://github.com/clitetailor/cursornext/blob/master/docs/02-writing-test.md). You can also use the [`capture`](https://github.com/clitetailor/cursornext/blob/d4b1ccaca8b638ff06b0df216921d346539bf421/src/test.ts#L58-L142) method as a reference.

**Comment by arshwaraich12** [comment]
Hi, @clitetailor i would like to work on this. Just to confirm, this function is to be implemented in `src/test.ts` for `CursorTest` class, right?

**Comment by clitetailor** [comment]
> Hi, @clitetailor i would like to work on this. Just to confirm, this function is to be implemented in src/test.ts for CursorTest class, right?

Yes, it is. It is a feature for generating test case.

> Also, is this functionality the opposite of the capture method?

Not really, actually it's the same as `capture`. Currently, there're two kinds of syntax. The inline one:

```
🌵(cursor)Hello, 🌵(target)World!
This is an example of how caret diagram and caret syntax work.
```

Which can be used as:

```ts
const { cursor, target } = t.inline`
🌵(cursor)Hello, 🌵(target)World!
This is an example of how caret diagram and caret syntax work.
`
```

And the block style one (or you can use any other name that suitable):

```
1 | Hello, World!
  | ^      ^
  | |      target
  | cursor
2 | This is an example of how caret diagram and caret syntax work.
```

Which produce the same result:

```ts
const { cursor, target } = t.block`
    1 | Hello, World!
      | ^      ^
      | |      target
      | cursor
    2 | This is an example of how caret diagram and caret syntax work.
`
```

Currently, we already have the [inline style function](https://github.com/clitetailor/cursornext/blob/5bd16c17c82ee1ab86250331ff8cd72b0d12e303/src/test.ts#L83-L174). The purpose of this issue is to support the block style syntax. And yes this is a quite complex issue but also the interesting one.

My first thought is to convert the block syntax into inline syntax because i can't think of any nice way to make the tagged template receives a string input also:

```ts
t.capture('Hello, World!🌵')

t.inline`Hello, World🌵`
```

But for now, I think that returning a `CaptureResult` directly is much more appropriate for this issue:

```ts
const { cursor, target } = t.block`
    1 | Hello, World!
      | ^      ^
      | |      target
      | cursor
    2 | This is an example of how caret diagram and caret syntax work.
`
```

You can choose any of the either ways to finish the issue. What do you think?

**Comment by arshwaraich12** [comment]
I feel like returning a `CaptureResult` would be more consistent with the rest of the design. I will work on that, thanks.

**Comment by clitetailor** [comment]
Hi @arshwaraich12, is there any progress on this issue?

**Comment by clitetailor** [comment]
Solved in #12
